posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,560 @@
+
+## What makes Serverless and GraphQL such a great fit?
+
+When moving to GraphQL, you suddenly rely on one HTTP endpoint to connect your clients to your backend services. Once you do decide to do that, you want this one HTTP endpoint to:
+
+1. be auto-scaling
+2. be reliable
+3. be fast
+4. have a small attack vector regarding security
+
+All these properties are fulfilled by a single AWS Lambda function in combination with API Gateway. It’s just a great fit! 
+
+In sum, powering your GraphQL endpoint with a serverless backend solves scaling and availability concerns outright, and it gives you a big leg up on security. It’s not even much code nor configuration. It takes only a few minutes to get to a production ready setup.
+
+I'd recommend you read [Jared's](https://twitter.com/ShortJared) [post](https://www.trek10.com/blog/a-look-at-serverless-graphql/) to get a better understanding of this relationship ;)
+
+## Serverless-GraphQL repository
+
+It’s pretty straightforward to get your HTTP endpoint up and running.
+
+I am happy to announce our Open Source Initiatives with [nikgraf](https://twitter.com/nikgraf) in [Serverless and GraphQL Repository](https://github.com/serverless/serverless-graphql). There, I walk through all the steps in detail. Go check it out!
+
+![alt text](https://user-images.githubusercontent.com/1587005/36035218-1c06763c-0d6b-11e8-996b-996243b0975f.png "Serverless and GraphQL Architecture")
+
+Repository comes in two flavor —
+
+1. API Gateway and Lambda Backend 
+
+ - API Gateway handles HTTP requests and responses and invokes the Lambda function in response to HTTP requests. 
+ - You manage GraphQL server in AWS Lambda using [apollo-server-lambda](https://www.npmjs.com/package/apollo-server-lambda) 
+ - Lambda function retrieves data from DynamoDB, [REST backend](https://developer.twitter.com/en/docs), or RDS and returns it to the client via API Gateway.
+
+**Note:** We are currently working on adding more backend integrations including GraphCool Prisma, Druid, MongoDB, AWS Neptune, etc.
+
+2. AppSync Backend 
+
+ - AppSync GraphQL Proxy server handles HTTP requests and responses fully managed by AWS.
+ - It uses [VTL](http://velocity.apache.org/engine/1.7/vtl-reference.html) under the hood to transform GraphQL request and response.
+ - Built-in integrations with DynamoDB, Elastic Search, and AWS Lambda. 
+
+Overall, there isn’t much code nor configuration required. You can get this to a production-ready setup in a few minutes. 
+
+**Note:** In this blog, I am going to explore creating GraphQL endpoints using API Gateway and Lambda Backend. I will dedicate my next blog to build the same endpoints using AWS Appsync.
+
+## Let's create a Serverless GraphQL Endpoint
+
+The [serverless-graphql](https://github.com/serverless/serverless-graphql) repo has the full walkthrough; give it a try and let me know what you think.
+
+*Step 1: Configure Serverless Template*
```

Comment:
Everywhere you've delineated with italics or horizontal lines, we'll need to change to H3s or H4s for formatting consistency with previous blog posts.

Reply:
ok, we can come back to this at the end where you can help with consistency with previous posts.